commit 80cb961e5380a3878246d41341ff91378ca59e05
Author: Jia Tan <jiat0218@gmail.com>
Date:   Fri Aug 4 22:17:11 2023 +0800

    codespell: Add .codespellrc to set default options.
    
    The .codespellrc allows setting default options to avoid false positive
    matches, set additional dictionaries, etc. For now, codespell can be
    used locally before committing doc and comment changes.
    
    It should help prevent silly errors and fix up commits in the future.

diff --git a/.codespellrc b/.codespellrc
new file mode 100644
index 00000000..b2c11346
--- /dev/null
+++ b/.codespellrc
@@ -0,0 +1,24 @@
+[codespell]
+
+# Skip all translation files and a few other autogenerated files.
+# The autotool files should have their typos fixed in the upstream, but
+# until then we will blacklist them here.
+skip = *.po,*.pot,./po4a/man,./doc/api,./configure,./autom4te.cache,./m4/libtool.m4,./build-aux/depcomp,./build-aux/ltmain.sh,./build-aux/config.guess,./build-aux/config.rpath,./m4/po.m4,./build-aux/config.sub
+
+# Ignore false positive matching words. Ideally codespell would allow
+# ignoring words for specific files, but that does not appear to be
+# supported. Instead we need to hope we do not make these typos.
+# Additionally, the ignored words must be specified lower-case even though
+# some of the false positives only occurred upper-case.
+# ANS - used as a variable name in xzmore.in.
+# bu - groff syntax for creating a bullet list item, used in xz.1.
+# te - groff syntax, used in xz.1.
+# caf - command line options for tar example, used in xz.1.
+ignore-words-list = ans,bu,te,caf
+
+# Add extra dictionaries to help improvement comments, docs, etc.
+builtin = clear,rare,informal,usage,names
+
+# Always default to highest interactive level to avoid accidentally
+# changing a false positive or picking the wrong replacement.
+interactive = 3